src/libs/Network.js

```diff
@@ -161,6 +201,18 @@ function processNetworkRequestQueue() {
             .catch(error => onError(queuedRequest, error));
     });
 
+    // We should clear the NETWORK_REQUEST_QUEUE when we have loaded the persisted requests & they are processed.
+    // As multiple client will be sharing the same Queue and NETWORK_REQUEST_QUEUE is synchronized among clients,
+    // we only ask Leader client to clear the queue
+    if (ActiveClientManager.isClientTheLeader() && didLoadPersistedRequests) {
```

Comment:
NAB:  Do we just need `didLoadPersistedRequests` just for this line?

Reply:
So If you try to think the bigger scope about all the tabs that are open in the browser with E.cash then the `ActiveClientManager.isClientTheLeader()` check makes sense. Without it, every tab would be handling the Offline queue.

Imagine, you send a message `Hey` form One of the browser tab while you are offline.
You have 5 tabs opened on your browser with E.cash.
If you check all 5 you will see that your `hey ` message is visible but grayed out.
Now you come back online, What do you expect that all tab should send `Hey` to server. or it should be sent once.
If server receives 5 requests, you will see 5 `hey` messages on your chat thus we only ask Leader to send that message and other client will automatically be synced.

Hope it clears the doubt.

Reply:
Gotcha, but since that check happens in line 57 of `Network.js` isn't it redundant here?

Reply:
We only want one tab to clear the queue, there could be other clients there and if they all clear the queue the it makes prone to errors. And yes Onyx calls will be multiplied by a number of tabs.

Reply:
But even step #8 in the PR description say "When the user comes back online"

I'm just wondering how/when `Onyx.connect` gets called again.

Reply:
Cool this makes sense to me now, thanks!